gitclone/gitclone_test.go

```diff
@@ -113,7 +113,7 @@ var testCases = [...]struct {
 			Branch: "hcnarb",
 		},
 		wantCmds: []string{
-			`git "fetch" "--tags"`,
+			`git "fetch" "--tags" "origin" "refs/heads/hcnarb"`,
```

Comment:
Instead of this we may fetch only tags: https://stackoverflow.com/questions/1204190/does-git-fetch-tags-include-git-fetch

Reply:
I mean not in this scenario but when only tag is provided

Reply:
BTW I wouldn't fetch tags here only if the "all-tags" input is true

Reply:
In this PR I tried to change a single behavior of the step: each fetch should have a branch as refspec set (if branch is provided) so that fetches are limited to that branch.

What about adding this information to the related ticket (Git Clone Step: introduce an input for tag fetch control), I was not aware of this 🤔